registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -169,23 +167,46 @@ NormalDistributionsTransform<PointSource, PointTarget>::computeDerivatives (Eige
                                                                             const Eigen::Matrix<double, 6, 1> &transform,
                                                                             bool compute_hessian)
 {
-  score_gradient.setZero ();
-  hessian.setZero ();
-  double score = 0;
+  std::vector<double> scores(num_threads_, 0.0);
+  std::vector<Eigen::Matrix<double, 6, 1>, Eigen::aligned_allocator<Eigen::Matrix<double, 6, 1>>> score_gradients(num_threads_, Eigen::Matrix<double, 6, 1>::Zero().eval());
+  std::vector<Eigen::Matrix<double, 6, 6>, Eigen::aligned_allocator<Eigen::Matrix<double, 6, 6>>> hessians(num_threads_, Eigen::Matrix<double, 6, 6>::Zero().eval());
 
   // Precompute Angular Derivatives (eq. 6.19 and 6.21)[Magnusson 2009]
   computeAngleDerivatives (transform);
 
   // Update gradient and hessian for each point, line 17 in Algorithm 2 [Magnusson 2009]
-  for (std::size_t idx = 0; idx < input_->size (); idx++)
+  #pragma omp parallel for num_threads(num_threads_) schedule(guided, 8)
```

Comment:
Any reason for specifying the schedule?

Reply:
I tested different scheduling methods and empirically determined that guided scheduling with chunk size = 8 is fastest in this case. I added a note to describe this.